// app/(tabs)/index.tsx
import { Image, StyleSheet, Platform, ScrollView } from 'react-native';
import { HelloWave } from '@/components/HelloWave';
import ParallaxScrollView from '@/components/ParallaxScrollView';
import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';

export default function HomeScreen() {
  return (
    <ParallaxScrollView
      headerBackgroundColor={{ light: '#3498db', dark: '#1a5276' }}
      headerImage={
        <Image
          source={require('@/assets/images/partial-react-logo.png')}
          style={styles.reactLogo}
        />
      }>
      <ThemedView style={styles.titleContainer}>
        <ThemedText type="title">My iOS App</ThemedText>
        <HelloWave />
      </ThemedView>
      
      <ThemedView style={styles.sectionContainer}>
        <ThemedText type="subtitle">Welcome to Your iOS App</ThemedText>
        <ThemedText>
          This is a cross-platform mobile application built with React Native and Expo.
          It provides a modern, responsive interface with iOS-specific optimizations.
        </ThemedText>
      </ThemedView>
      
      <ThemedView style={styles.sectionContainer}>
        <ThemedText type="subtitle">Features</ThemedText>
        <ThemedView style={styles.featureItem}>
          <ThemedText type="defaultSemiBold">• Responsive Design</ThemedText>
          <ThemedText>Adapts perfectly to different iPhone screen sizes</ThemedText>
        </ThemedView>
        <ThemedView style={styles.featureItem}>
          <ThemedText type="defaultSemiBold">• Dark Mode Support</ThemedText>
          <ThemedText>Automatically switches between light and dark themes</ThemedText>
        </ThemedView>
        <ThemedView style={styles.featureItem}>
          <ThemedText type="defaultSemiBold">• Tab Navigation</ThemedText>
          <ThemedText>Easy navigation between different sections of the app</ThemedText>
        </ThemedView>
      </ThemedView>
      
      <ThemedView style={styles.sectionContainer}>
        <ThemedText type="subtitle">Get Started</ThemedText>
        <ThemedText>
          Explore the app by tapping on the Explore tab to discover more features and content.
        </ThemedText>
      </ThemedView>
    </ParallaxScrollView>
  );
}

const styles = StyleSheet.create({
  titleContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 8,
    marginBottom: 16,
  },
  sectionContainer: {
    gap: 12,
    marginBottom: 24,
    padding: 8,
    borderRadius: 8,
    borderWidth: 1,
    borderColor: '#e0e0e0',
  },
  featureItem: {
    marginBottom: 8,
  },
  reactLogo: {
    height: 178,
    width: 290,
    bottom: 0,
    left: 0,
    position: 'absolute',
  },
});
