// app/(tabs)/settings.tsx
import React from 'react';
import { StyleSheet, ScrollView } from 'react-native';
import { ThemedText } from '@/components/ThemedText';
import { ThemedView } from '@/components/ThemedView';
import UserProfileComponent from '@/components/UserProfile';
import ContentLibraryComponent from '@/components/ContentLibrary';
import NotificationsComponent from '@/components/Notifications';
import { Tabs, TabScreen } from '@/components/ui/Tabs';

export default function SettingsScreen() {
  return (
    <ScrollView style={styles.container}>
      <ThemedView style={styles.header}>
        <ThemedText type="title">Settings</ThemedText>
        <ThemedText>Manage your app preferences and profile</ThemedText>
      </ThemedView>
      
      <Tabs>
        <TabScreen title="Profile">
          <UserProfileComponent />
        </TabScreen>
        <TabScreen title="Content">
          <ContentLibraryComponent />
        </TabScreen>
        <TabScreen title="Notifications">
          <NotificationsComponent />
        </TabScreen>
      </Tabs>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  header: {
    padding: 16,
    gap: 8,
    marginBottom: 16,
  },
});
